AlwaysOn/Wearable/src/androidTest/java/com/example/android/wearable/wear/alwayson/MainActivityTests.kt

```diff
@@ -0,0 +1,215 @@
+import androidx.lifecycle.Lifecycle
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.core.app.launchActivity
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+import java.time.ZoneId
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@OptIn(ExperimentalCoroutinesApi::class)
+class MainActivityTests {
+    private val context = ApplicationProvider.getApplicationContext<Application>()
+
+    private val testDispatcher: TestCoroutineDispatcher = TestCoroutineDispatcher()
+
+    /**
+     * A timestamp in the relatively far, far future (year 2200).
+     *
+     * This ensures the real alarm manager won't actually trigger.
+     */
+    private var instant = Instant.ofEpochMilli(7258118400000)
+
+    private lateinit var scenario: ActivityScenario<MainActivity>
+
+    private lateinit var uiDevice: UiDevice
+
+    @Before
+    fun setup() {
+        uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+        // Ensure we are starting in active mode
+        uiDevice.pressKeyCode(KeyEvent.KEYCODE_WAKEUP)
+
+        // Override the active dispatcher with a paused test one.
+        testDispatcher.pauseDispatcher()
+        activeDispatcher = testDispatcher
+
+        updateClock()
+        scenario = launchActivity()
+    }
+
+    @After
+    fun teardown() {
+        scenario.close()
+    }
+
+    @Test
+    fun initialTextIsCorrect(): Unit = runBlocking {
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        onView(withId(R.id.time)).check(matches(withText("00:00:00")))
+        onView(withId(R.id.time_stamp)).check(matches(withText(context.getString(R.string.timestamp_label, 7258118400000L))))
+        onView(withId(R.id.state)).check(matches(withText(context.getString(R.string.mode_active_label))))
+        onView(withId(R.id.draw_count)).check(matches(withText(context.getString(R.string.draw_count_label, 1))))
+    }
+
+    @Test
+    fun textIsCorrectAfterFiveSeconds(): Unit = runBlocking {
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        // Advance 5 seconds, one at a time
+        repeat(5) {
+            advanceTime(Duration.ofSeconds(1))
+        }
+
+        onView(withId(R.id.time)).check(matches(withText("00:00:05")))
```

Comment:
Could you make this a constant? You use it other locations too.

Reply:
Done, I added constants for each of the time displays.